lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1073,9 +1080,11 @@ FIRRTLModuleLowering::lowerModule(FModuleOp oldModule, Block *topLevelModule,
     newModule->setAttr("firrtl.extract.cover.extra", builder.getUnitAttr());
 
   // If this is in the test harness, make sure it goes to the test directory.
+  // Do not update output file information if it is already present.
   if (auto testBenchDir = loweringState.getTestBenchDirectory())
     if (loweringState.isInTestHarness(oldModule)) {
-      newModule->setAttr("output_file", testBenchDir);
+      if (!newModule->hasAttr("output_file"))
```

Comment:
When might it already be set -- is this something new with this change?

Reply:
This was trying to future proof the conversion. I wasn't aware of this being a known problem though.

High level: it seems like the existing `output_file` attribute should be respected even if we don't have an example of a pass hitting this edge case. If it's not, then that's likely a bug. The other place that an `output_file` is set is on binds and that will _not_ overwrite a pre-existing `output_file`.

Reply:
Sounds perfect, thanks for the consideration and explanation! :rocket: :100: